### PR TITLE
Kr/expert body apis

### PR DESCRIPTION
**Pull request by KanishkaRajputd**
## Release Notes

- **New Features**
	- Introduced functionality to distinguish expert comments with a new parameter.
	- Added an API handler for users to submit expert requests.
	- New API handler to check if a user address is recognized as an expert.
	- Enhanced comment structure to include an indication of expert status.

- **Bug Fixes**
	- Improved error handling for post ID validation and data fetching.

- **Documentation**
	- Updated message entries for better clarity on expert requests.

- **Refactor**
	- Expanded type definitions to include expert request statuses.

### CODE REVIEW

File: pages/api/v1/expertBody/becomeExpert.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+const handler: NextApiHandler<MessageType> = async (req, res) => {
+	storeApiKeyUsage(req);
+
+	try {
+		const network = String(req.headers['x-network']);
+		if (!network || !isValidNetwork(network)) return res.status(400).json({ message: messages.INVALID_NETWORK });
+
+		const { contribution, userAddress, reason } = req.body;
+
+		const encodedUserAddress = getEncodedAddress(userAddress, network);
```
Comment (alphainfinitus) [inline pages/api/v1/expertBody/becomeExpert.ts:26]: check if this address belongs to the user or not

File: pages/api/v1/expertBody/becomeExpert.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+			.where('address', '==', encodedUserAddress)
+			.get();
+
+		if (!expertReqDocs.empty) {
+			return res.status(400).json({ message: messages.EXPERT_REQ_ALREADY_EXIST });
+		}
+
+		const expertReqDoc = expertReqSnapshot.doc();
+
+		const payload = {
```
Comment (alphainfinitus) [inline pages/api/v1/expertBody/becomeExpert.ts:55]: please store updatedAt and approvedAt etc as well

File: pages/api/v1/posts/on-chain-post.ts [hunk 3]
```diff
@@ -393,6 +393,7 @@ export async function getComments(
 						comment_source: 'polkassembly',
 						content: '[Deleted]',
 						created_at: data.created_at?.toDate ? data.created_at.toDate() : data.created_at,
+						expertComment: data?.expertComment || false,
```
Comment (alphainfinitus) [inline pages/api/v1/posts/on-chain-post.ts:396]: should be `isExpertComment`

**Review by alphainfinitus — CHANGES_REQUESTED** [review]

File: pages/api/v1/expertBody/becomeExpert.ts (hunk 1 above)
Comment (KanishkaRajputd) [reply pages/api/v1/expertBody/becomeExpert.ts:26]: ok

**Pull request merged**